linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_password_pam_pwhistory_remember_system_auth/ansible/shared.yml

```diff
@@ -5,55 +5,219 @@
+        authselect_current_profile: "{{ result_authselect_profile.stdout }}"
+        authselect_custom_profile: "custom/hardening"
+      when:
+        - result_authselect_profile is not skipped
+        - result_authselect_profile.stdout is not match("custom/")
+
+    - name: Get authselect current features to also enable them in the custom profile
+      ansible.builtin.shell:
+        cmd: authselect current | tail -n+3 | awk '{ print $2 }'
+      register: result_authselect_features
+      changed_when: false
+      when:
+        - result_authselect_profile is not skipped
+        - authselect_current_profile is not match("custom/")
+
+    - name: Check if any custom profile with the same name was already created in the past
+      ansible.builtin.stat:
+        path: /etc/authselect/{{ authselect_custom_profile }}
+      register: result_authselect_custom_profile_present
+      changed_when: false
+      when:
+        - authselect_current_profile is not match("custom/")
+
+    - name: Create a custom profile based on the current profile
+      ansible.builtin.command:
+        cmd: authselect create-profile hardening -b sssd
+      register: result_authselect_create_profile
+      when:
+        - result_authselect_check_cmd is success
+        - authselect_current_profile is not match("custom/")
+        - not result_authselect_custom_profile_present.stat.exists
+
+    - name: Ensure the desired remember value is updated in the custom profile
+      ansible.builtin.replace:
+        dest: "/etc/authselect/{{ authselect_custom_profile }}/system-auth"
+        regexp: '(.*pam_pwhistory.so.*remember=)(\S+)(.*)$'
+        replace: '\g<1>{{ var_password_pam_remember }}\g<3>'
+      when:
+        - result_authselect_profile is not skipped
+        - authselect_custom_profile is match("custom/")
+        - result_pam_pwhistory_present.found == 1
+        - result_pam_pwhistory_remember_present.found == 1
+
+    - name: Ensure the remember parameter is included in the custom profile
+      ansible.builtin.replace:
+        dest: "/etc/authselect/{{ authselect_custom_profile }}/system-auth"
+        regexp: '(.*pam_pwhistory.so.*)(?! remember=\S+)(.*)$'
+        replace: '\g<1> \g<2> remember={{ var_password_pam_remember }}'
+      when:
+        - result_authselect_profile is not skipped
+        - authselect_custom_profile is match("custom/")
+        - result_pam_pwhistory_present.found == 1
+        - result_pam_pwhistory_remember_present.found == 0
+
+    - name: Ensure the desired control value is updated in the custom profile
+      ansible.builtin.replace:
+        dest: "/etc/authselect/{{ authselect_custom_profile }}/system-auth"
+        regexp: '^(password\s+)((?:(?:requisite)|(?:required)))(\s+pam_pwhistory\.so\s.*)$'
+        replace: '\g<1>{{ var_password_pam_remember_control_flag }}\g<3>'
+      when:
+        - result_authselect_profile is not skipped
+        - authselect_custom_profile is match("custom/")
```

Comment:
Is `authselect_custom_profile is match("custom/")` necessary?
At this point `authselect_custom_profile` certainly starts with "custom/".